client/signup/steps/site/index.jsx

```diff
@@ -186,7 +187,7 @@ module.exports = React.createClass( {
 	},
 
 	getErrorMessagesWithLogin( fieldName ) {
-		const link = login( { isNative: true, redirectTo: window.location.href } ),
+		const link = login( { isNative: config.isEnabled( 'login/native-login-links' ), redirectTo: window.location.href } ),
```

Comment:
We should update the [same method](https://github.com/Automattic/wp-calypso/blob/0316d6b7197255581fb903c9fa784e578c27454a/client/components/signup-form/index.jsx#L287) in the `<SignupForm />` component as well.